lib/3scale_toolbox/commands/import_command/openapi/resource_reader.rb

```diff
@@ -0,0 +1,50 @@
+module ThreeScaleToolbox
+  module Commands
+    module ImportCommand
+      module OpenAPI
+        module ResourceReader
+          ##
+          # Reads resources from different types of sources.
+          # Supported types are: file, URL, stdin
+          # Return type is
+          # [content, format] where
+          # content: raw content
+          # format: Hash with single key: `format`. Value can be `:json` or `:yaml`
+          # format example: { format: :json }
+          def openapi_resource(resource)
+            case resource
+            when '-'
+              method(:read_stdin)
+            when /\A#{URI::DEFAULT_PARSER.make_regexp}\z/
+              method(:read_url)
+            else
+              method(:read_file)
+            end.call(resource)
+          end
+
+          # Detect format from file extension
+          def read_file(resource)
+            [File.read(resource), { format: File.extname(resource) }]
+          end
+
+          def read_stdin(_resource)
+            content = STDIN.read
```

Comment:
I think you can collapse all these implementations into `YAML.parse(ARGF)`

Reply:
First, as mentioned in https://github.com/3scale/3scale_toolbox/pull/76#discussion_r241045479, we still need to find out format.

Second, ARGF is like `cat`. Nice but what if we would like to add some  extra positional param?  Less flexible as far as I see.

Reply:
Format is not an issue as already mentioned there.

What extra positional arguments? Do you have an example?

It really depends what CRI does to the ARGV and if leaves unprocessed arguments.
That is not as big a deal anyway.

Reply:
according to https://robots.thoughtbot.com/rubys-argf
```
ARGF will interpret all elements of the ARGV array as filenames and when read will produce a concatenation of the contents of these files. If ARGV is empty, then ARGF reads from standard input.
```
Commands and subcommands will be treated as filenames, so hard to integrate with CRI command framework. `ARGV` needs to be processed first. Moreover, options and flags need to be removed from `ARGV` before calling `ARGF.read`.

WRT extra positional argument, imagine we want to pass 3scale instance remote name as positional argument instead of option with `-d`. It would be treated like filename or it would need to be removed before calling `ARGF.read`.

IMO unnecessary complication for the provided little value.